Inline mirror://hackage and update URLs

mgorny opened the pull request:
Since `mirror://hackage` has only one URL, inline it in all ebuilds. While at it, also update to the modern hackage URLs (the old ones have permanent redirect to the new ones).

mgorny commented:
In case anybody asked, I've verified the new URIs for all packages. There are 404s on other URIs in those packages but hackage URIs are all fine.

mgorny commented:
@gentoo/haskell, ping.

mgorny commented:
(I'll rebase once I get an ack)

trofi commented on dev-lang/ghc/ghc-7.10.3.ebuild:
```diff
@@ -18,23 +18,23 @@ inherit autotools bash-completion-r1 eutils flag-o-matic ghc-package
+arch_binaries="$arch_binaries ppc? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-ppc.tbz2 )"
+arch_binaries="$arch_binaries ppc64? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-ppc64.tbz2 )"
+arch_binaries="$arch_binaries sparc? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PV}-sparc.tbz2 )"
+arch_binaries="$arch_binaries x86? ( http://slyfox.uni.cx/~slyfox/distfiles/ghc-bin-${PVR}-x86.tbz2 )"
```
This does the reverse from what is stated in commit message. `https://slyfox.uni.cx` should just work.

mgorny replied:
I do wonder why I did that. Maybe it had some temporary SSL problems. I'll revert it.

trofi commented on app-admin/haskell-updater/haskell-updater-1.2.10.ebuild:
```diff
@@ -8,7 +8,7 @@ inherit eutils haskell-cabal
 DESCRIPTION="Rebuild Haskell dependencies in Gentoo"
 HOMEPAGE="https://wiki.haskell.org/Gentoo#haskell-updater"
-SRC_URI="mirror://hackage/packages/archive/${PN}/${PV}/${P}.tar.gz"
+SRC_URI="https://hackage.haskell.org/package/${PN}-${PV}/${P}.tar.gz"
```
`${PN}-${PV}/${P}.tar.gz` looks confusing here. I don't see why not `${P}/${P}.tar.gz`

mgorny replied:
I went for the minimal change in my sed, didn't think of actually replacing that. But'll do.

trofi commented on dev-lang/ghc/ghc-7.10.3.ebuild:
```diff
@@ -182,7 +182,7 @@ update_SRC_URI() {
 		pn=$1 pv=$2
 
-		SRC_URI+=" mirror://hackage/package/${pn}/${pn}-${pv}.tar.gz"
+		SRC_URI+=" https://hackage.haskell.org/package/${pn}/${pn}-${pv}.tar.gz"
```
This is not consistent to other changes. The rest of URL mangling in pull request does the equivalent of `${pn}-${pv}/${pn}-${pv}.tar.gz` and not `${pn}/${pn}-${pv}.tar.gz`. They should agree.

mgorny replied:
Yeah, my grep+sed wnet for `packages` but this one had `package` instead. Will improve.

mgorny commented:
I suppose I can commit just the ebuild updates for now and keep the tpm entry until everything else is ready. The main idea is to avoid regressions.

mgorny commented:
Ok, rebased and fixed all the issues. I'm going to update the commit message tomorrow — obviously, my .bash_history has all the seds prior to the right one but the one that handed the slash-to-dash transition isn't there. Anyway, I'll probably write something better then to cover PN/PV while at it.

mgorny commented:
Ping.

mgorny commented:
Oh, I didn't get to do the commit message thing yet.
